src/main/scala/com/fulcrumgenomics/bam/TrimPrimers.scala

```diff
@@ -0,0 +1,195 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.fulcrumgenomics.bam
+
+import com.fulcrumgenomics.FgBioDef._
+import com.fulcrumgenomics.bam.SamRecordClipper.ClippingMode
+import com.fulcrumgenomics.cmdline.{ClpGroups, FgBioTool}
+import com.fulcrumgenomics.util.{DelimitedDataParser, Io}
+import dagr.commons.util.LazyLogging
+import dagr.sopt.{arg, clp}
+import htsjdk.samtools.SAMFileHeader.SortOrder
+import htsjdk.samtools.SamPairUtil.PairOrientation
+import htsjdk.samtools._
+import htsjdk.samtools.util._
+import Math.abs
+
+import scala.collection.JavaConversions.iterableAsScalaIterable
+
+@clp(group=ClpGroups.SamOrBam, description=
+  """
+    |Trims primers from reads post-alignment.  Takes in a BAM file of aligned reads
+    |and a tab-delimited file with five columns (chrom, left_start, left_stop, right_start,
+    |and right_stop) which provide the 1-based inclusive start and end positions of the
+    |primers for each amplicon.
+    |
+    |Paired end reads that map to a given amplicon position are trimmed so that the
+    |alignment no-longer includes the primer sequences.
+    |
+    |All other aligned reads have the maximum primer length trimmed!
```

Comment:
Probably a good idea to state that sorting will occur if the input is not `queryname` sorted.